### DISCUSSION TITLE
Why do we use regularization for Kernel SHAP?

### DISCUSSION THREAD

**Discussion by ssiu** [shap/shap#2481] [Q&A]
In figure 3 of the paper it says that Kernel SHAP is used with debiased lasso - but it's proved in Theorem 2 that the regularization term should be zero. So why does it make sense to add regularization in practice?

**Comment by jzking** [comment]
My uninformed guesses:

- It reduces the importance of variables that are multi co-linear or perfectly correlated such that a single variable is weighted the highest while the others are minimized towards 0 importance
- It helps to solve the least squares equation in solve() faster and/or increases the likelihood of convergence

EDIT:
Looking into this further, section 4.1 of https://arxiv.org/pdf/1705.07874.pdf provides some information on regularization and states finding the correct regularization parameter is necessary to calculate shapley values.  I'd be lying if I truly understood why, the answer is buried in the mathematical relationship between LIME and SHAP used in the kernel method it seems.

**Reply by c-foschi** [reply]
Section 4.1 of that paper states that the correct regularization parameter is 0.
